/* Copyright 2023 The FedLearner Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package fedlearner_webconsole.proto;


message Metric {
    // training or validation steps
    repeated double steps = 1;
    // metric values
    repeated double values = 2;
}

message ConfusionMatrix {
  int64 tp = 1;
  int64 tn = 2;
  int64 fp = 3;
  int64 fn = 4;
}

message ModelJobMetrics {
  map<string, Metric> train = 1;
  map<string, Metric> eval = 2;
  ConfusionMatrix confusion_matrix = 3;
  map<string, float> feature_importance = 4;
}
